### ISSUE TITLE
Memory assignment error in EncodedFile.readinto

### ISSUE THREAD

**Issue by flying-sheep** [althonos/pronto#109]
Your EncodedFile class’s `readinto` can’t handle if `codecs.StreamRecoder.read` returns a bigger chunk than requested:

https://github.com/althonos/pronto/blob/37126293472293b4bb3f406b41198a351fec4542/pronto/utils/io.py#L55-L62

But this can happen:

<img width="398" alt="image" src="https://user-images.githubusercontent.com/291575/101761778-d8d53680-3adc-11eb-8cb9-9fd0b8b68e7b.png">

And that’s normal, see the [docs for StreamReader](https://docs.python.org/3/library/codecs.html#codecs.StreamReader) (emphasis mine):

> The size argument indicates the *approximate maximum number* of encoded bytes or code points to read for decoding. [...]

The consequence:
```pytb
  File "test.py", line 2
    ont = pronto.Ontology('some/path.owl')
  File pronto/ontology.py", line 217, in __init__
    cls(self).parse_from(_handle)  # type: ignore
  File pronto/parsers/rdfxml.py, line 86, in parse_from
    tree: etree.ElementTree = etree.parse(handle)
  File xml/etree/ElementTree.py, line 1202, in parse
    tree.parse(source, parser)
  File xml/etree/ElementTree.py, line 595, in parse
    self._root = parser._parse_whole(source)
  File pronto/utils/io.py, line 24, in read
    return super(BufferedReader, self).read(size)
  File pronto/utils/io.py, line 61, in readinto
    typing.cast(bytearray, buffer)[: len(chunk)] = chunk
ValueError: memoryview assignment: lvalue and rvalue have different structures
```

**Comment by flying-sheep** [comment]
Yeah, this is a bit more difficult than the others I found, but with #110 merged less people should run into it.

Kinda an edge case I guess, but still worth fixing.

**Comment by althonos** [comment]
The weird part here is that even the standard library seems to assume `read` does return less than the requested amount of data (e.g. the [`BufferedIOBase.readinto`](https://github.com/python/cpython/blob/master/Modules/_io/bufferedio.c#L81) generic implementation).

As a fudge, the most direct fix is to wrap the given file into a `BufferedReader`, but there is always a risk that decoding/encoding will take more space, for instance on decoding a `latin-1` text into `UTF-8`...

**Comment by flying-sheep** [comment]
Hmm, maybe we should ask a core dev about this.

**Comment by althonos** [comment]
I'll release the patch you made, we can keep this for a later release. As you said, there are chances it doesn't really matter if Unicode is being used directly.

**Comment by flying-sheep** [comment]
Thank you! Yes, it only matters if the `utf-8` branch isn’t hit.

**Comment by althonos** [comment]
I removed this bit of `EncodedFile`, a direct copy should be fast enough (and in most cases unneeded since we can assume files to be `utf-8` encoded, and `v2.7.0` allows controlling the encoding).
